main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -2206,7 +2206,7 @@ object Weeder {
           case (x :: xs, Some(Pattern.Wild(l))) => WeededAst.Pattern.Record(x :: xs, Pattern.Wild(l), loc).toSuccess
 
           // Bad Pattern { | r }
-          case (Nil, Some(r)) => Validation.toHardFailure(EmptyRecordExtensionPattern(r.loc))
+          case (Nil, Some(r)) => Validation.toSoftFailure(r, EmptyRecordExtensionPattern(r.loc))
```

Comment:
Maybe this could turn into an ascription to enforce that it's _some_ record type

Reply:
No ascriptions in patterns :/